CHANGELOG.md

```diff
@@ -2,6 +2,142 @@
 
+## [0.12.0](https://github.com/rerun-io/rerun/compare/0.11.0...0.12.0)
+
+### Overview & Highlights
+- 🌁 The Rerun Viewer now supports a plugin system for creating arbitrary external data loaders:
+  - See the [guide for creating external data loaders](https://www.rerun.io/docs/howto/open-any-file).
+- 🕸️ The viewer now fetches examples from the web, featuring many new built-in examples.
+- 🐍 The Python SDK now works with Python-3.12.
+- 📘 Blueprint containers can now be selected and modified.
+- 🚀 Space Views are now evaluated in parallel for improved performance.
+
+### Details
+
+#### 🌊 C++ API
+- Exposing `recording_id` in C and C++ SDKs [#4384](https://github.com/rerun-io/rerun/pull/4384)
+- All C++ preprocessor macros start now with RR_ (instead of a mix of RR_ and RERUN_) [#4371](https://github.com/rerun-io/rerun/pull/4371)
+- C++ & Python API: add helpers for constructing an entity path [#4595](https://github.com/rerun-io/rerun/pull/4595)
+
+#### 🐍 Python API
+- Add `--stdout`/`-o` to our CLI helper library [#4544](https://github.com/rerun-io/rerun/pull/4544)
+- Make it possible to log segmentation images directly as floats [#4585](https://github.com/rerun-io/rerun/pull/4585)
+- C++ & Python API: add helpers for constructing an entity path [#4595](https://github.com/rerun-io/rerun/pull/4595)
+- Python SDK: introduce deferred garbage collection queue [#4583](https://github.com/rerun-io/rerun/pull/4583)
+- Add support for Python 3.12 [#4146](https://github.com/rerun-io/rerun/pull/4146)
+
+#### 🦀 Rust API
+- Exposing `recording_id` in Rust SDK [#4383](https://github.com/rerun-io/rerun/pull/4383)
+- Add `--stdout`/`-o` to our CLI helper library [#4544](https://github.com/rerun-io/rerun/pull/4544)
+- Document how to construct an entity path for the Rust logging API [#4584](https://github.com/rerun-io/rerun/pull/4584)
+
+#### 🪳 Bug Fixes
+- Bugfix: show labels on segmentation images with trivial dimensions [#4368](https://github.com/rerun-io/rerun/pull/4368)
+- Datastore: don't eagerly sort in bucket split routine on ingestion path [#4417](https://github.com/rerun-io/rerun/pull/4417)
+- Resolve spurious blueprint panel group collapsing [#4548](https://github.com/rerun-io/rerun/pull/4548)
+- Fix rectangle that indicates the zoomed pixel area on hover being one pixel to small [#4590](https://github.com/rerun-io/rerun/pull/4590)
+- Fix wrong RowId order of logged data [#4658](https://github.com/rerun-io/rerun/pull/4658)
+- Make scroll-to-zoom a lot more responsive in 3D views [#4668](https://github.com/rerun-io/rerun/pull/4668)
+- Fix heuristic object properties being broken in some cases / fix DepthMeter being ignored sometimes [#4679](https://github.com/rerun-io/rerun/pull/4679)
+
+#### 🌁 Viewer Improvements
+- Make viewer contexts's render context reference non-mutable [#4430](https://github.com/rerun-io/rerun/pull/4430)
+- Pass viewer context always non-mutable [#4438](https://github.com/rerun-io/rerun/pull/4438)
```

Comment:
more of a refactor, should move down
changelog gen script might need adjustment; these are tagged as refactors